packages/fast-usdc/src/fast-usdc.contract.js

```diff
@@ -44,21 +47,56 @@ export const contract = async (zcf, privateArgs, zone, tools) => {
   assert('PoolShares' in terms.brands, 'no PoolShares brand');
 
   const statusManager = prepareStatusManager(zone);
-  const feed = prepareTransactionFeed(zone);
   const makeSettler = prepareSettler(zone, { statusManager });
   const { chainHub, vowTools } = tools;
   const makeAdvancer = prepareAdvancer(zone, {
     chainHub,
-    feed,
     log: trace,
     statusManager,
     vowTools,
   });
-  assertAllDefined({ feed, makeAdvancer, makeSettler, statusManager });
+  const makeFeedKit = prepareTransactionFeedKit(zone);
+  assertAllDefined({ makeFeedKit, makeAdvancer, makeSettler, statusManager });
+  const feedKit = makeFeedKit();
+  const advancer = makeAdvancer(
+    // @ts-expect-error FIXME
+    {},
+  );
+  // Connect evidence stream to advancer
+  void observeIteration(subscribeEach(feedKit.public.getEvidenceStream()), {
+    updateState(evidence) {
+      try {
+        advancer.handleTransactionEvent(evidence);
+      } catch (err) {
+        trace('🚨 Error handling transaction event', err);
+      }
+    },
+  });
 
   const creatorFacet = zone.exo('Fast USDC Creator', undefined, {});
 
-  return harden({ creatorFacet });
+  const publicFacet = zone.exo('Fast USDC Public', undefined, {
+    // XXX to be removed before production
+    /**
+     * NB: Any caller with access to this invitation maker has the ability to
+     * add evidence.
+     *
+     * Provide an API call in the form of an invitation maker, so that the
+     * capability is available in the smart-wallet bridge.
+     *
+     * @param {CctpTxEvidence} evidence
+     */
+    async makeTestPushInvitation(evidence) {
+      await feedKit.admin.addEvidence(evidence);
```

Comment:
> why change state in the invitationMaker instead of the offerHandler

https://github.com/Agoric/agoric-sdk/pull/10452#discussion_r1838493353

>If addEvidence were to throw, will smart-wallet get the error message?

Good question. The throw would happen on [this line](https://github.com/Agoric/agoric-sdk/blob/eb1d655e6b33c0044138e95867ae8ad527115de3/packages/smart-wallet/src/smartWallet.js#L917) so it would be an [IMMEDIATE_OFFER_ERROR](https://github.com/Agoric/agoric-sdk/blob/master/packages/smart-wallet/src/smartWallet.js#L958-L959) and show up as [an error in the wallet status](https://github.com/Agoric/agoric-sdk/blob/master/packages/smart-wallet/src/smartWallet.js#L968-L971).

I added a todo to test that when we integrate with smart-wallet in bootstrap tests

Reply:
Ok cool, I see now. 

> I added a todo to test that when we integrate with smart-wallet in bootstrap tests

🙏